app/src/main/java/fr/gaulupeau/apps/Poche/service/BGService.java

```diff
@@ -124,402 +110,232 @@ protected void onHandleIntent(Intent intent) {
-        } finally {
-            daoSession.getDatabase().endTransaction();
-        }
-
-        if(queueChangedLength != null) {
-            postEvent(new OfflineQueueChangedEvent(queueChangedLength));
-        }
-
-        Log.d(TAG, "deleteArticle() finished");
-        return result;
-    }
-
-    private ActionResult deleteArticleRemote(int articleID) {
-        ActionResult result = null;
-
-        try {
-            if(!getWallabagService().deleteArticle(articleID)) {
-                result = new ActionResult(ActionResult.ErrorType.NegativeResponse);
-            }
-        } catch(RequestException | IOException e) {
-            ActionResult r = processException(e, "deleteArticleRemote()");
-            if(!r.isSuccess()) result = r;
+            Log.d(TAG, "syncOfflineQueue() finished processing queue item");
         }
 
-        return result;
-    }
-
-    private ActionResult addLink(ActionRequest actionRequest) {
-        String link = actionRequest.getLink();
-        Log.d(TAG, String.format("addLink(%s) started", link));
-
-        ActionResult result = new ActionResult();
-        Long queueChangedLength = null;
-
-        DaoSession daoSession = getDaoSession();
-        daoSession.getDatabase().beginTransaction();
-        try {
-            QueueHelper queueHelper = new QueueHelper(daoSession);
+        Long queueLength = null;
 
-            if(queueHelper.addLink(link)) {
-                if(WallabagConnection.isNetworkAvailable()) {
-                    result.updateWith(addLinkRemote(link));
-                } else {
-                    result.setErrorType(ActionResult.ErrorType.NoNetwork);
-                }
-
-                if(!result.isSuccess()) {
-                    queueHelper.enqueueAddLink(link);
-                }
+        if(!completedQueueItems.isEmpty()) {
+            daoSession.getDatabase().beginTransactionNonExclusive();
+            try {
+                queueHelper.dequeueItems(completedQueueItems);
 
-                Log.d(TAG, "addLink() synced: " + result.isSuccess());
-            } else {
-                Log.d(TAG, "addLink(): QueueHelper reports there's nothing to do");
-            }
+                queueLength = queueHelper.getQueueLength();
 
-            if(queueHelper.isQueueChanged()) {
-                queueChangedLength = queueHelper.getQueueLength();
+                daoSession.getDatabase().setTransactionSuccessful();
+            } finally {
+                daoSession.getDatabase().endTransaction();
             }
-
-            daoSession.getDatabase().setTransactionSuccessful();
-        } finally {
-            daoSession.getDatabase().endTransaction();
         }
 
-        if(queueChangedLength != null) {
-            postEvent(new OfflineQueueChangedEvent(queueChangedLength));
+        if(queueLength != null) {
+            postEvent(new OfflineQueueChangedEvent(queueLength));
+        } else {
+            queueLength = (long)queueItems.size();
```

Comment:
I cast `int` to `long` so the compiler could implicitly cast it to `Long`. You can't directly cast `int` to `Long`, though.

You can try `Long.valueOf(queueItems.size())`, but Android Studio will suggest to replace it with `(long)`. You can also use `(Long)(long)`, but that's just redundant.